### PR TITLE
Upgrade deprecated v1alpha1 DomainMapping API to newer one

### LINKED ISSUE #1854: Upgrade deprecated v1alpha1 DomainMapping API to newer one
The upgrade is failing due to missing v1alpha1 DomainMapping API. We need to upgrade to newer version. 

https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative_client/1852/unit-tests_client_main/1693436714506260480#

_Originally posted by @dsimansk in https://github.com/knative/client/issues/1852#issuecomment-1686227886_

### PR DESCRIPTION
**Pull request by xiangpingjiang**
## Reference

Fixes #1854 

**Release Note**

```release-note
🐣 Upgrade deprecated `v1alpha1` DomainMapping API to `v1beta1`
```

### CODE REVIEW

File: pkg/kn/commands/completion_helper.go [hunk 1]
```diff
@@ -223,7 +223,7 @@ func completeDomain(config *completionConfig) (suggestions []string) {
 	if err != nil {
 		return
 	}
-	client, err := config.params.NewServingV1alpha1Client(namespace)
+	client, err := config.params.NewServingv1beta1Client(namespace)
```
[the remark this answers is not in the document]
Comment (dsimansk) [reply pkg/kn/commands/completion_helper.go:226]: It should be fixed at the original defition. Better not to merge the suggestion. :)

**Review by dsimansk — CHANGES_REQUESTED** [review]

File: pkg/kn/commands/completion_helper.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (xiangpingjiang) [reply pkg/kn/commands/completion_helper.go]: done

**Review by dsimansk — APPROVED** [review]
Thanks @xiangpingjiang!

**Pull request merged**